src/modules/tokens/services/token.service.ts

```diff
@@ -139,29 +110,71 @@ export class TokenService {
     @ErrorLoggerAsync({
         logArgs: true,
     })
-    @GetOrSetCache({
-        baseKey: 'token',
-        remoteTtl: CacheTtlInfo.Token.remoteTtl,
-        localTtl: CacheTtlInfo.Token.localTtl,
-    })
-    async tokenMetadata(tokenID: string): Promise<EsdtToken> {
-        return this.tokenMetadataRaw(tokenID);
-    }
+    async tokenMetadata(
+        tokenID: string,
+        fields?: (keyof EsdtToken)[],
+    ): Promise<EsdtToken> {
+        // TODO: add caching
+        const projection: ProjectionType<EsdtTokenDocument> = {};
```

Comment:
You don't need this method. You can use `getAllTokensMetadeta` with a single tokenID